turbopack/crates/turbopack-core/src/chunk/chunk_group.rs

```diff
@@ -315,6 +315,76 @@ pub async fn chunk_group_content(
         },
     )?;
 
+    // This needs to use the unmerged items
+    let availability_info = availability_info
+        .with_modules(Vc::cell(state.chunkable_items.clone()))
+        .await?;
+
+    if should_merge_modules {
+        let merged_modules = module_graph.merged_modules().await?;
+        state.chunkable_items = state
+            .chunkable_items
+            .into_iter()
+            .map(async |chunkable_module| match chunkable_module {
+                ChunkableModuleOrBatch::Module(module) => {
+                    if !merged_modules.should_create_chunk_item_for(ResolvedVc::upcast(module)) {
+                        return Ok(smallvec![]);
+                    }
+
+                    let module = if let Some(replacement) =
+                        merged_modules.should_replace_module(ResolvedVc::upcast(module))
+                    {
+                        replacement
+                    } else {
+                        module
+                    };
+
+                    Ok(smallvec![ChunkableModuleOrBatch::Module(module)])
+                }
+                ChunkableModuleOrBatch::Batch(batch) => {
+                    let batch_ref = batch.await?;
+                    let modules = &batch_ref.modules;
+
+                    let modified = AtomicBool::new(false);
```

Comment:
can't this just be a `mut bool`?   iirc `filter` and `map` take `Fnmut` so that they can mutate in scope variables.

If it needs to be AtomicBool you should probably use `Ordering::Relaxed` since there are no threads involved.

Reply:
Yes, they are `FnMut`, but the are both live at the same time, so cannot both take mutably borrow the local variable.
Changed to `Relaxed`